### PR TITLE
fix: fix issue with forking

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Fix issue with forking

Fixes https://github.com/appsmithorg/appsmith/issues/35835

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10715600770>
> Commit: 9a2678c8d590846054bf693d032a6b61fcc31471
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10715600770&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Thu, 05 Sep 2024 08:15:19 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced the Editor component to better track page transitions, improving performance and user experience.
  
- **Bug Fixes**
  - Streamlined logic for handling page ID changes to prevent unnecessary re-initialization of the editor when navigating back to the default page.

### CODE REVIEW

**Comment by sagar-qa007** [comment]
@brayn003 Should we consider updating an existing test case or adding a new one for unit testing?
Your insights would be appreciated.

**Comment by brayn003** [comment]
@sagar-qa007 think we need to add brand new test cases for this. I will pick this up as a separate task

File: app/client/src/pages/Editor/index.tsx [hunk 1]
```diff
@@ -111,6 +113,21 @@ class Editor extends Component<Props> {
...
+
+    const pageId = this.props.pages.find(
+      (page) => page.basePageId === basePageId,
+    )?.pageId;
+
+    const prevPageId = prevProps.pages.find(
+      (page) => page.basePageId === prevPageBaseId,
+    )?.pageId;
+    // caching value for prevPageId as it is required in future lifecycles
+    if (!!prevPageId) {
```
[the remark this answers is not in the document]
Comment (mohanarpit) [reply app/client/src/pages/Editor/index.tsx:125]: @brayn003 Any reason we resolved this comment? For my understanding, is the double negation still required?

[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/pages/Editor/index.tsx:125]: @mohanarpit not any particular reason, just a little opinionated on using double negations. Also, helps when debugging with breakpoint

[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/pages/Editor/index.tsx]: Removed the double negation

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**